### PR TITLE
fq 0.9.1

### PR DESCRIPTION
**Pull request by guoxiao**

### CODE REVIEW

File: Library/Formula/fq.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  depends_on "concurrencykit"
+  depends_on "jlog"
+  depends_on "openssl"
+
+  def install
+    inreplace "Makefile", "/usr/lib/dtrace", "#{lib}/dtrace"
+    system "make", "PREFIX=#{prefix}"
+    system "make", "install", "PREFIX=#{prefix}"
+    bin.install "fqc", "fq_sndr", "fq_rcvr"
```
Comment (mikemcquaid) [inline Library/Formula/fq.rb]: Why aren't these installed by `make install`?

File: Library/Formula/fq.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  test do
+    system "fqd", "-h"
```
Comment (mikemcquaid) [inline Library/Formula/fq.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by mikemcquaid** [comment]
The test is failing: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/29897/version=yosemite/testReport/junit/brew-test-bot/yosemite/test_fq/

File: Library/Formula/fq.rb (hunk 1 above)
Comment (guoxiao) [inline Library/Formula/fq.rb]: The `make install` only does the install job. We must run `make` first.

**Comment by guoxiao** [comment]
@mikemcquaid Are there any details of the error? I've tested on my machine and it works well.

**Comment by dunn** [comment]
It's possible you just need to replace `system "fqd", "-h"` with `system bin/"fqd", "-h"`.

**Comment by guoxiao** [comment]
@mikemcquaid @dunn Oh I found the `fqd` is placed in `sbin` folder, that's why it's not found.

Should the `/usr/local/sbin` folder be added to `PATH` by default?

**Comment by DomT4** [comment]
> Oh I found the fqd is placed in sbin folder, that's why it's not found.

`system sbin/"fqd", "-h"`, then.

> Should the /usr/local/sbin folder be added to PATH by default?

`system bin/` and `system sbin/` are shorthand for `system /usr/local/opt/formula/bin/executable` and the same for `sbin`. Nothing is tested directly from `HOMEBREW_PREFIX`.

**Comment by guoxiao** [comment]
I want to find the public IPv4 Address of the machine, so I use the following
```ruby
address = Socket.ip_address_list.find { |a| a.ipv4? && !a.ipv4_loopback? }.ip_address
```
But it doesn't work in ML. What's the better way to do this?

**Comment by guoxiao** [comment]
@mikemcquaid @dunn @DomT4 Any suggestions about the problem?
I'm not very familiar with ruby and I don't have a ML machine to test on.

**Comment by xu-cheng** [comment]
Why you need the public IP address? Why not just `127.0.0.1`?

File: Library/Formula/fq.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    inreplace "Makefile", "/usr/lib/dtrace", "#{lib}/dtrace"
+    system "make", "PREFIX=#{prefix}"
+    system "make", "install", "PREFIX=#{prefix}"
+    bin.install "fqc", "fq_sndr", "fq_rcvr"
+  end
+
+  test do
+    require "socket"
+    address = Socket.ip_address_list.find { |a| a.ipv4? && !a.ipv4_loopback? }.ip_address
+    system sbin/"fqd", "-h"
```
Comment (xu-cheng) [inline Library/Formula/fq.rb]: This line can be removed. Because below test is sufficient enough.

**Comment by guoxiao** [comment]
@xu-cheng Thanks for your help. I came up with a better way to solve the problem.
The `-n` option of `fqd` doesn't accept `127.0.0.1` or `0.0.0.0`, so I thought I must get the public address of machine. However just now I found I can simply omit the `-n`, and it's listening on `127.0.0.1`.

**Comment by xu-cheng** [comment]
Thanks for the pull request! 🎉 Homebrew depends on contributions from community members like you and we're grateful for your support.

**Pull request closed without merging**